src/amr/data/field/coarsening/electric_field_coarsener.hpp

```diff
@@ -0,0 +1,159 @@
+#ifndef PHARE_FLUX_SUM_COARSENER
+#define PHARE_FLUX_SUM_COARSENER
+
+#include "core/data/grid/gridlayoutdefs.hpp"
+#include "core/utilities/constants.hpp"
+#include "amr/resources_manager/amr_utils.hpp"
+
+
+#include <SAMRAI/hier/Box.h>
+#include <stdexcept>
+
+namespace PHARE::amr
+{
+using core::dirX;
+using core::dirY;
+using core::dirZ;
+/** @brief This class gives an operator() that performs the coarsening of N fine nodes onto a
+ * given coarse node
+ *
+ * A MagneticFieldCoarsener object is created each time the refine() method of the
+ * FieldCoarsenOperator is called and its operator() is called for each coarse index.
+ * It is the default coarsening policy and used for any field that does not come with
+ * specific constraints (such as conserving some property in the coarsening process).
+ *
+ *
+ * This coarsening operation is defined so to conserve the magnetic flux.
+ * This is done by assigning to a magnetic field component on a coarse face, the average
+ * of the enclosed fine faces
+ *
+ */
+template<std::size_t dimension>
+class ElectricFieldCoarsener
+{
+public:
+    ElectricFieldCoarsener(std::array<core::QtyCentering, dimension> const centering,
+                           SAMRAI::hier::Box const& sourceBox,
+                           SAMRAI::hier::Box const& destinationBox,
+                           SAMRAI::hier::IntVector const& ratio)
+        : centering_{centering}
+        , sourceBox_{sourceBox}
+        , destinationBox_{destinationBox}
+
+    {
+    }
+
+    template<typename FieldT>
+    void operator()(FieldT const& fineField, FieldT& coarseField,
+                    core::Point<int, dimension> coarseIndex)
+    {
+        // For the moment we only take the case of field with the same centering
+        TBOX_ASSERT(fineField.physicalQuantity() == coarseField.physicalQuantity());
+
+        core::Point<int, dimension> fineStartIndex;
```

Comment:
`
core::Point<int, dimension> fineStartIndex{generate([&](auto const& coarse){
 return coarsecoarse * this->ratio_;}, *coarseIndex)}
`

maybe simpler